### PR TITLE
Fixed error and warnings.

### PR DESCRIPTION
**Pull request by lee-emqx**

### CODE REVIEW

File: include/nng/protocol/mqtt/mqtt_parser.h [hunk 1]
```diff
@@ -11,11 +11,11 @@
 // Do not change to %lu! just supress the warning of compiler!
 #define DISCONNECT_MSG          \
 	"{\"username\":\"%s\"," \
-	"\"ts\":%llu,\"reason_code\":\"%x\",\"client_id\":\"%s\"}"
+	"\"ts\":%lu,\"reason_code\":\"%x\",\"client_id\":\"%s\"}"
 
 #define CONNECT_MSG                                                           \
 	"{\"username\":\"%s\", "                                              \
-	"\"ts\":%llu,\"proto_name\":\"%s\",\"keepalive\":%d,\"return_code\":" \
```
Comment (JaylinYu) [inline include/nng/protocol/mqtt/mqtt_parser.h:18]: I made this deliberately, intend to be compatible with ARM32

Comment (JaylinYu) [reply include/nng/protocol/mqtt/mqtt_parser.h:18]: please remove this change

File: include/nng/protocol/mqtt/mqtt_parser.h [hunk 2]
```diff
@@ -11,11 +11,11 @@
 // Do not change to %lu! just supress the warning of compiler!
 #define DISCONNECT_MSG          \
 	"{\"username\":\"%s\"," \
-	"\"ts\":%llu,\"reason_code\":\"%x\",\"client_id\":\"%s\"}"
```
Comment (JaylinYu) [inline include/nng/protocol/mqtt/mqtt_parser.h:14]: please remove this change

File: src/nng.c [hunk 3]
```diff
@@ -65,7 +65,7 @@ nng_zalloc(size_t sz)
 	return (nni_zalloc(sz));
 }
 
-void *
+void
```
Comment (JaylinYu) [inline src/nng.c:68]: sure this wont break anything?

**Review by JaylinYu — CHANGES_REQUESTED** [review]

File: include/nng/protocol/mqtt/mqtt_parser.h (hunk 1 above)
Comment (lee-emqx) [reply include/nng/protocol/mqtt/mqtt_parser.h]: That's change removed.

Comment (alvin1221) [reply include/nng/protocol/mqtt/mqtt_parser.h]: I prefer using `cJSON` to format json string.

**Review by alvin1221 — APPROVED** [review]

**Pull request merged**